### PR TITLE
mgr/cephadm: guard _check_daemons

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Continue with other daemons, if one fails

Fixes: https://tracker.ceph.com/issues/46748
Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -2084,12 +2078,19 @@ def _check_daemons(self):
...
+                            host=dd.hostname,
+                            daemon_id=dd.daemon_id,
+                            daemon_type=dd.daemon_type),
+                        reconfig=True)
+                except OrchestratorError as e:
+                    self.events.from_orch_error(e)
+                    # continue...
+                except Exception as e:
+                    self.events.for_daemon_from_exception(dd.name(), e)
+                    # continue...
```
Comment (mgfritch) [inline src/pybind/mgr/cephadm/module.py:2093]: looks mostly good, but should we allow the daemon post actions to occur after `_create_daemon` fails ... ?

Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/module.py:2093]: sigh. Thanks!

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2020-07-30_14:25:43-rados:cephadm-wip-swagner-testing-2020-07-30-1435-distro-basic-smithi/

* just some environment issues.

**Comment by sebastian-philipp** [comment]
green: https://pulpito.ceph.com/swagner-2020-08-04_10:27:26-rados:cephadm-wip-swagner3-testing-2020-08-04-1030-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
jenkins test make check

**Comment by sebastian-philipp** [comment]
```
=================================== FAILURES ===================================
______________________ TestCephadm.test_daemon_check_post ______________________

self = <cephadm.tests.test_cephadm.TestCephadm object at 0x7f0cc46c65c0>
cephadm_module = <cephadm.module.CephadmOrchestrator object at 0x7f0cc469f438>

    @mock.patch("cephadm.module.CephadmOrchestrator._run_cephadm", _run_cephadm('{}'))
    def test_daemon_check_post(self, cephadm_module: CephadmOrchestrator):
        with with_host(cephadm_module, 'test'):
            with with_service(cephadm_module, ServiceSpec(service_type='grafana'), CephadmOrchestrator.apply_grafana, 'test'):
    
                # Make sure, _check_daemons does a redeploy due to monmap change:
                cephadm_module._store['_ceph_get/mon_map'] = {
                    'modified': datetime.datetime.utcnow().strftime(CEPH_DATEFMT),
                    'fsid': 'foobar',
                }
                cephadm_module.notify('mon_map', None)
    
                with mock.patch("cephadm.module.CephadmOrchestrator.mon_command") as _mon_cmd:
    
                    cephadm_module._check_daemons()
>                   _mon_cmd.assert_any_call({'prefix': 'dashboard set-grafana-api-url', 'value': 'https://test:3000'})

cephadm/tests/test_cephadm.py:263: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <MagicMock name='mon_command' id='139692269904952'>
args = ({'prefix': 'dashboard set-grafana-api-url', 'value': 'https://test:3000'},)
kwargs = {}
expected = (({'prefix': 'dashboard set-grafana-api-url', 'value': 'https://test:3000'},), {})
actual = [], cause = None
expected_string = "mon_command({'prefix': 'dashboard set-grafana-api-url', 'value': 'https://test:3000'})"

    def assert_any_call(self, *args, **kwargs):
        """assert the mock has been called with the specified arguments.
    
        The assert passes if the mock has *ever* been called, unlike
        `assert_called_with` and `assert_called_once_with` that only pass if
        the call is the most recent one."""
        expected = self._call_matcher((args, kwargs))
        actual = [self._call_matcher(c) for c in self.call_args_list]
        if expected not in actual:
            cause = expected if isinstance(expected, Exception) else None
            expected_string = self._format_mock_call_signature(args, kwargs)
            raise AssertionError(
                '%s call not found' % expected_string
>           ) from cause
E           AssertionError: mon_command({'prefix': 'dashboard set-grafana-api-url', 'value': 'https://test:3000'}) call not found

/usr/lib/python3.6/unittest/mock.py:876: AssertionError
```

**Comment by sebastian-philipp** [comment]
green: https://pulpito.ceph.com/swagner-2020-08-05_13:20:00-rados:cephadm-wip-swagner-testing-2020-08-05-1302-distro-basic-smithi/

**Pull request merged**